### PR TITLE
detect the detach operation and propogate it

### PR DESCRIPTION
**Pull request by caglar10ur**
This will be used for on-demand serial work later...

/cc @sflxn

### CODE REVIEW

File: lib/apiservers/engine/backends/container.go [hunk 1]
```diff
@@ -534,9 +534,7 @@ func (c *Container) containerStart(name string, hostConfig *containertypes.HostC
 	if bind {
 		e := c.findPortBoundNetworkEndpoint(hostConfig, endpoints)
 		if err = c.mapPorts(portmap.Map, hostConfig, e); err != nil {
-			err = fmt.Errorf("error mapping ports: %s", err)
-			log.Error(err)
```
Comment (fdawg4l) [inline lib/apiservers/engine/backends/container.go]: This looked somewhat useful.  Is the err still being logged somewhere?

Comment (caglar10ur) [inline lib/apiservers/engine/backends/container.go]: Hmm not really. I'll move that logging into the mapPorts function though as the rest of the errors gets logged there

Comment (caglar10ur) [inline lib/apiservers/engine/backends/container.go]: Hmm wait, we log it when we get the InternalServer error on the other side like the rest of the errors.

File: lib/apiservers/engine/backends/container.go [hunk 2]
```diff
@@ -1119,36 +1133,53 @@ func attachStreams(ctx context.Context, vc *viccontainer.VicContainer, clStdin i
...
 
-	log.Infof("container attach:  cleaned up connections to %s.", vc.ContainerID)
+	// close the channel so that we don't leak (if there is an error)/or get blocked (if there are no errors)
+	close(errors)
+
+	log.Infof("cleaned up connections to %s. Checking errors", vc.ContainerID)
 	for err := range errors {
 		if err != nil {
+			// check if we got DetachError
+			if _, ok := err.(DetachError); ok {
```
Comment (fdawg4l) [inline lib/apiservers/engine/backends/container.go]: Will we get a `DetachError` from all of the `go func`s above or is it possible only 1 returns this error, while the other 2 return CDE?  Just asking out of curiosity.

Comment (caglar10ur) [inline lib/apiservers/engine/backends/container.go]: Nope, we are only intercepting stdin

File: lib/apiservers/engine/backends/container.go (hunk 1 above)
Comment (fdawg4l) [inline lib/apiservers/engine/backends/container.go]: Cool.  Thanks!

**Comment by fdawg4l** [comment]
LGTM

**Comment by dougm** [comment]
lgtm

**Comment by caglar10ur** [comment]
♻️

**Pull request merged**